.github/workflows/release-branch-tests.yml

```diff
@@ -0,0 +1,156 @@
+# **what?**
+# The purpose of this workflow is to trigger CI to run for each
+# release branch and main branch in a reusable format. If the CI workflow
+# fails for a branch, it will post to dev-core-alerts to raise awareness.
+
+# **why?**
+# Ensures release branches and main are always shippable and not broken.
+# Also, can catch any dependencies shifting beneath us that might
+# introduce breaking changes (could also impact Cloud).
+
+# **when?**
+# Mainly on a schedule of 9:00, 13:00, 18:00 UTC everyday.
+# Manual trigger can also test on demand
+
+name: Release branch scheduled testing
+
+on:
+  workflow_call:
+    inputs:
+      workflows_to_run:
+        description: 'A list of workflow(s) to run aginst ex: ["main.yml, "integration.yml"]'
+        required: true
+        type: string
+
+# no special access is needed
+permissions: read-all
+
+jobs:
+  prep-work:
+    name: "Audit Inputs"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "[DEBUG] Print Inputs"
+        run: |
+          echo "workflows_to_run:    ${{ inputs.workflows_to_run }}"
+
+  fetch-latest-branches:
+    runs-on: ubuntu-latest
+
+    outputs:
+      branches_to_test: ${{ steps.add-main.outputs.full_branch_list }}
+
+    steps:
+      - name: "Fetch ${{ github.event.repository.name }} Latest Branches"
+        uses: dbt-labs/actions/fetch-repo-branches@v1.1.1
+        id: get-latest-branches
+        with:
+          repo_name: ${{ github.event.repository.name }}
+          organization: "dbt-labs"
+          pat: ${{ secrets.GITHUB_TOKEN }}
+          fetch_protected_branches_only: true
+          regex: "^1.[0-9]+.latest$"
+          perform_match_method: "match"
+          retries: 3
+
+      - name: "Add main to branch list"
+        id: add-main
+        run: |
+          full_branch_list="${{ steps.get-latest-branches.outputs.repo-branches }}"
+          # Swap single and double quotes because bash doesnt interpret correctly
+          full_branch_list="${full_branch_list//\'/__SINGLE_QUOTE__}"
+          full_branch_list="${full_branch_list//\"/\'}"
+          full_branch_list="${full_branch_list//__SINGLE_QUOTE__/\"}"
+          # now convert it to a bash array and add the main branch
+          full_branch_list=($(echo "$full_branch_list" | jq -r '.[]'))
+          for item in "${full_branch_list[@]}"; do   echo "$item"; done
+          full_branch_list+=("main")
```

Comment:
I don't know that it would simplify this much as the string itself is in the format "['first', 'second']" so I'd have to modify it to append to it anyway.  The real solution is to have the `fetch-repo-branches` actions be able to return something like "supported branches" that includes main.  Once we start dropping support for older versions we're going to have to do this.  That would completely remove this step.